src/google/protobuf/parse_context.h

```diff
@@ -1196,7 +1196,14 @@ template <typename Add, typename SizeCb>
 const char* EpsCopyInputStream::ReadPackedVarint(const char* ptr, Add add,
                                                  SizeCb size_callback) {
   int size = ReadSize(&ptr);
-  size_callback(size);
+  auto element_size = 0;
+  const uint8_t* pData = (uint8_t*)ptr;
+  for (auto i = 0; i < size; i++) {
+    if (pData[i] < 0x80) {
+      element_size++;
+    }
+  }
+  size_callback(element_size);
```

Comment:
unfortunately, the type of size_callback is specified as bytes and we depend on this fact elsewhere.  

https://github.com/protocolbuffers/protobuf/blob/461b50b18097122f8a5ab3c3686f8254827a1716/src/google/protobuf/generated_message_tctable_lite.cc#L1344-L1366